### PR TITLE
Move convertToTensorValueType from TFPartition to TFUtilities

### PR DESCRIPTION
**Pull request by bgogul**
Moves the convertToTensorValueType helper function from TFPartition.cpp to TFUtilities.

### CODE REVIEW

**Comment by bgogul** [comment]
@swift-ci please test tensorflow

**Comment by rxwei** [comment]
@swift-ci please clean test tensorflow

**Review by rxwei — APPROVED** [review]

**Review by lattner — APPROVED** [review]

File: lib/SILOptimizer/Mandatory/TFUtilities.h [hunk 1]
```diff
@@ -239,6 +239,14 @@ struct GraphGlobalConfiguration {
   /// LLVM Builtin type like Builtin.f32) into the TensorFlow TF_DataType value.
   unsigned convertSwiftTypeToTF(Type ty);
 
+  /// `ty` must be a valid TensorFlow element type "T", like Builtin.Int32. Turn
+  /// it into a TensorHandle<T> type.
+  SILType convertToTensorValueType(Type ty, const ASTContext& ctx);
```
Comment (mhong) [inline lib/SILOptimizer/Mandatory/TFUtilities.h]: I feel having the overloaded names here could cause potential confusion. How about rename to sth like: convertElementTypeToTensorValueType()?

(now that we are moving this method to a more public API, it'd be useful to scrutinize the naming a bit more)

**Review by mhong — APPROVED** [review]

File: lib/SILOptimizer/Mandatory/TFUtilities.h (hunk 1 above)
Comment (bgogul) [reply lib/SILOptimizer/Mandatory/TFUtilities.h]: Good point. Changed to convertElementTypeToTensorValueType.

**Comment by bgogul** [comment]
@swift-ci please test tensorflow

**Pull request merged**